app/src/main/java/com/duckduckgo/app/location/ui/SiteLocationPermissionDialog.kt

```diff
@@ -0,0 +1,176 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.location.ui
+
+import android.app.Dialog
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.global.faviconLocation
+import com.duckduckgo.app.global.image.GlideApp
+import com.duckduckgo.app.global.view.gone
+import com.duckduckgo.app.global.view.website
+import com.duckduckgo.app.location.data.LocationPermissionType
+import org.jetbrains.anko.find
+
+class SiteLocationPermissionDialog : DialogFragment() {
+
+    interface SiteLocationPermissionDialogListener {
+        fun onSiteLocationPermissionSelected(domain: String, permission: LocationPermissionType)
+    }
+
+    val listener: SiteLocationPermissionDialogListener
+        get() {
+            return if (parentFragment is SiteLocationPermissionDialogListener) {
+                parentFragment as SiteLocationPermissionDialogListener
+            } else {
+                activity as SiteLocationPermissionDialogListener
+            }
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        isCancelable = false
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val rootView = layoutInflater.inflate(R.layout.content_site_location_permission_dialog, null)
+
+        val title = rootView.find<TextView>(R.id.sitePermissionDialogTitle)
+        val favicon = rootView.find<ImageView>(R.id.sitePermissionDialogFavicon)
+        val allowAlways = rootView.find<TextView>(R.id.siteAllowAlwaysLocationPermission)
+        val allowOnce = rootView.find<TextView>(R.id.siteAllowOnceLocationPermission)
+        val denyOnce = rootView.find<TextView>(R.id.siteDenyOnceLocationPermission)
+        val denyAlways = rootView.find<TextView>(R.id.siteDenyAlwaysLocationPermission)
+        val extraDivider = rootView.find<View>(R.id.siteAllowOnceLocationPermissionDivider)
+        val anotherDivider = rootView.find<View>(R.id.siteDenyLocationPermissionDivider)
+
+        val alertDialog = AlertDialog.Builder(requireActivity(), R.style.AlertDialogTheme)
+            .setView(rootView)
+
+        validateBundleArguments()
+        populateTitle(title)
+        populateFavicon(favicon)
+        configureListeners(allowAlways, allowOnce, denyOnce, denyAlways)
+        hideExtraViews(allowOnce, denyOnce, extraDivider, anotherDivider)
+        makeCancellable()
+
+        return alertDialog.create()
+    }
+
+    private fun populateTitle(title: TextView) {
+        arguments?.let { args ->
+            val originUrl = args.getString(KEY_REQUEST_ORIGIN)
+            val dialogTitle = getString(R.string.preciseLocationSiteDialogTitle, originUrl.website())
```

Comment:
`nullable` string lint warning here

Reply:
we know it can't be `null` as it's part of the `instance` call so we are safe to use `!!` here